osu.Framework/Graphics/Sprites/SpriteIcon.cs

```
@@ -120,6 +120,40 @@ public bool Shadow
             }
         }
 
+        private Color4 shadowColour = new Color4(0f, 0f, 0f, 0.2f);
+
+        /// <summary>
+        /// The colour of the shadow displayed around the icon. A shadow will only be displayed if the <see cref="Shadow"/> property is set to true.
+        /// </summary>
+        public Color4 ShadowColour
+        {
+            get => shadowColour;
+            set
+            {
+                shadowColour = value;
+
+                if (spriteShadow != null)
+                    spriteShadow.Colour = shadowColour;
```

The idea here is right, but this is usually done [by checking `IsLoaded` instead](https://github.com/ppy/osu-framework/blob/fa15cbee7992b7be1803042cedee233485003ee2/osu.Framework/Graphics/Containers/ModelBackedDrawable.cs#L47-L48), and having `LoadComplete()` [apply any values that were potentially set beforehand](https://github.com/ppy/osu-framework/blob/fa15cbee7992b7be1803042cedee233485003ee2/osu.Framework/Graphics/Containers/ModelBackedDrawable.cs#L91). ([Another game-side example.](https://github.com/ppy/osu/blob/7654df94f6f37b8382be7dfcb4f674e03bd35427/osu.Game.Tournament/Screens/Gameplay/Components/MatchHeader.cs))

Doing it that way is theoretically safer, because BDL executes on a separate thread rather than on the update thread, so with this implementation you could potentially have a race between the read of `spriteShadow` here and its write in BDL, which would cause a value set to be potentially silently dropped..

---

Thanks for the explanation. If I am understanding correctly, would that also mean that [`bool Shadow`](https://github.com/ppy/osu-framework/blob/a5852bd26f0b4a1b21d1d9b9854e871432d4302c/osu.Framework/Graphics/Sprites/SpriteIcon.cs#L118-L119) has the same issue. Should I include a change for that aswell?

---

Sure, I suppose. Probably never going to be a better time than now.